doc/deployment-schema.json

```diff
@@ -0,0 +1,343 @@
+                                    "examples": [
+                                        "1.0"
+                                    ]
+                                },
+                                "routes": {
+                                    "type": "object",
+                                    "patternProperties": {
+                                        "^.+$": {
+                                            "type": "string",
+                                            "examples": [
+                                                "FROM /* INTO $upstream"
+                                            ]
+                                        }
+                                    }
+                                },
+                                "storeAndForwardConfiguration": {
+                                    "type": "object",
+                                    "required": [
+                                        "timeToLiveSecs"
+                                    ],
+                                    "properties": {
+                                        "timeToLiveSecs": {
+                                            "type": "integer",
+                                            "examples": [
+                                                7200
+                                            ]
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            },
+            "patternProperties": {
+                "^.+$": {
+                    "type": "object",
+                    "required": [
+                        "properties.desired"
+                    ],
+                    "properties": {
+                        "properties.desired": {
+                            "type": "object"
+                        }
+                    }
+                }
+            },
+            "additionalProperties": false
+        }
+    },
+    "definitions": {
+        "moduleType": {
+            "enum": [
+                "docker"
+            ]
+        },
+        "status": {
+            "enum": [
+                "running",
+                "stopped"
+            ]
+        },
+        "restartPolicy": {
+            "enum": [
+                "never",
+                "on-failure",
+                "on-unhealthy",
+                "always"
+            ]
+        },
+        "env": {
+            "type": "object",
+            "patternProperties": {
+                "^.+$": {
+                    "type": "object",
+                    "required": [
+                        "value"
+                    ],
+                    "properties": {
+                        "value": {
+                            "type": "string",
+                            "examples": [
+                                "UpstreamProtocol"
```

Comment:
Super nit: but the example might be confusing.. upstreamProtocol is typically the key..